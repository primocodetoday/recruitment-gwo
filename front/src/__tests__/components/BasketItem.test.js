import React from 'react';
import { render } from '@testing-library/react';
import { BasketItem } from 'components';
import { OrderContextProvider } from 'context/OrderContext';

describe('<BasketItem/> renders properly', () => {
  it('with passed data', () => {
    const props = {
      id: 34,
      title: 'Nasza Szkapa',
      quantity: 1,
      price: 2000,
    };

    const { getByText, container } = render(
      <OrderContextProvider>
        <BasketItem key={props.id} title={props.title} quantity={props.quantity} price={props.price} id={props.id} />,
      </OrderContextProvider>,
    );
    expect(getByText('Nasza Szkapa')).toBeTruthy();
    expect(getByText('20,00 zł')).toBeTruthy();
    expect(getByText('1')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
