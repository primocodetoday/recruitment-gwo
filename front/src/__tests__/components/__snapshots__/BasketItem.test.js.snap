// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasketItem/> renders properly with passed data 1`] = `
<li
  class="mb-3 px-3 rounded border list-group-item list-group-item-secondary"
>
  <div
    class=" align-items-center row"
  >
    <div
      class="col-md-8 col-sm-6 col-5"
    >
      <p
        class="h6 text-left"
      >
        Nasza Szkapa
      </p>
    </div>
    <div
      class="col-md-1 col-sm-2 col-2"
    >
      <p
        class="h6 text-center"
      >
        1
      </p>
    </div>
    <div
      class="col-md-2 col-sm-2 col-3"
    >
      <p
        class="h6 text-center"
      >
        20,00
         z≈Ç
      </p>
    </div>
    <div
      class=" d-flex justify-content-center col-md-1 col-sm-2 col-2"
    >
      <button
        class="text-center btn btn-danger btn-sm"
        type="button"
      >
        <i
          class="fas fa-times "
        />
      </button>
    </div>
  </div>
</li>
`;
